nakhll_market/serializers.py

```diff
@@ -645,6 +645,15 @@ class Meta:
             'MobileNumber': {'validators': []}
         }
 
+    def validate_NationalCode(self, value):
+        if not value:
+            raise serializers.ValidationError(
+                'وارد کردن کد ملی الزامی است'
+            )
+        elif len(value) != 10:
+            raise serializers.ValidationError('کد ملی باید 10 رقم باشد')
+        return value
```

Comment:
national code must be digit only!
```suggestion
        elif value.isdigit():
            raise serializers.ValidationError('کد ملی باید رقم باشد')
        return value
```

Reply:
now it is just a number

Reply:
> national code must be digit only!

It's better to store national code on the string to check it with regex later.

Reply:
and this was only the first step of validation.
it should be completed later.

Reply:
> > national code must be digit only!
> 
> It's better to store national code on the string to check it with regex later.

Ok, isdigit only checks that string is numbers only or not, it's a business rule, you know we don't have alphabetic chars in national code in the real world.

Reply:
> and this was only the first step of validation. it should be completed later.

I don't get the point, what do you mean?
Do you want to improve validation later?
Do you know dirty data can make enhancements painful ?!

Reply:
> > and this was only the first step of validation. it should be completed later.
> 
> I don't get the point, what do you mean? Do you want to improve validation later? Do you know dirty data can make enhancements painful ?!

I am going to add the validation of the national code tomorrow.